api/handlers.go

```diff
@@ -35,22 +39,29 @@ func (api *API) userHandlerGET(w http.ResponseWriter, req *http.Request, ps http
 }
 
 // userHandlerPOST creates a new user.
-func (api *API) userHandlerPOST(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
-	if err := req.ParseMultipartForm(MaxMultipartMem); err != nil {
-		WriteError(w, errors.New("Failed to parse multipart parameters."), http.StatusBadRequest)
+func (api *API) userHandlerPOST(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	email, err := database.NewEmail(req.PostFormValue("email"))
+	if err != nil {
+		WriteError(w, err, http.StatusBadRequest)
 		return
 	}
-	email := (user.Email)(req.PostFormValue("email"))
-	if !email.Validate() {
-		WriteError(w, user.ErrInvalidEmail, http.StatusBadRequest)
+	pw := req.PostFormValue("password")
+	if len(pw) == 0 {
+		WriteError(w, errors.New("The password cannot be empty."), http.StatusBadRequest)
 		return
 	}
-	u := &user.User{
+	u := &database.User{
 		FirstName: req.PostFormValue("firstName"),
 		LastName:  req.PostFormValue("lastName"),
 		Email:     email,
+		Tier:      database.TierFree,
+	}
+	err = u.SetPassword(pw)
+	if err != nil {
+		WriteError(w, errors.AddContext(err, "failed to set password"), http.StatusInternalServerError)
+		return
 	}
-	err := api.staticDB.UserCreate(req.Context(), u)
+	err = api.staticDB.UserCreate(req.Context(), u)
```

Comment:
F/U I'd change this to `UserCreate(ctx, firstname, lastname, email, tier, password)` for safety and convenience. It shouldn't be necessary to manually create users outside of the database package.

Reply:
F/U: #12